datafusion/physical-expr/src/expressions/get_indexed_field.rs

```diff
@@ -126,7 +138,32 @@ impl GetIndexedFieldExpr {
         start: Arc<dyn PhysicalExpr>,
         stop: Arc<dyn PhysicalExpr>,
     ) -> Self {
-        Self::new(arg, GetFieldAccessExpr::ListRange { start, stop })
+        Self::new(
+            arg,
+            GetFieldAccessExpr::ListStride {
+                start,
+                stop,
+                stride: Arc::new(Literal::new(ScalarValue::Int64(Some(1))))
+                    as Arc<dyn PhysicalExpr>,
+            },
+        )
+    }
+
+    /// Create a new [`GetIndexedFieldExpr`] for accessing the stride
+    pub fn new_stride(
```

Comment:
it is unused in this PR. `new_range` is also only used in test, which may be removed in the future since we can test them in slt.